great_expectations/datasource/data_connector/util.py

```diff
@@ -237,7 +237,7 @@ def convert_batch_identifiers_to_data_reference_string_using_regex(
 
 # noinspection PyUnresolvedReferences
 def _invert_regex_to_data_reference_template(
-    regex_pattern: str,
+    regex_pattern: Union[re.Pattern, str],
```

Comment:
Nit - let's use `re.Pattern | str`

Reply:
I'd rather we don't introduce new unions if we don't need them - why can't we just use a pattern here?

Reply:
@cdkini Done -- comitted; thanks.